scripts/setup/upgrade-postgres

```diff
@@ -43,12 +43,22 @@ pg_ctlcluster "$UPGRADE_FROM" main stop
     rm -rf "$TEMP_CONF_DIR"
 )
 
-pg_upgradecluster "$UPGRADE_FROM" main
+# Capture the output so we know where the path to the post-upgrade scripts is
+UPGRADE_LOG=$(mktemp)
+pg_upgradecluster "$UPGRADE_FROM" main --method=upgrade --link | tee "$UPGRADE_LOG"
+SCRIPTS_PATH=$(grep -o "/var/log/postgresql/pg_upgradecluster-$UPGRADE_FROM-$UPGRADE_TO-main.*" "$UPGRADE_LOG")
+rm "$UPGRADE_LOG"
```

Comment:
We should probably have error handling that doesn't remove the `UPGRADE_LOG` in the event that parsing the scripts path fails, given we don't have `set -x` here?

(Or do we need to remove it at all?)

Reply:
We do have `set -x`; so execution stops there, with an upgraded postgres but zulip not set to use it.  We should probably attempt to recover from that, yes.